src/serve/server.ts

```diff
@@ -0,0 +1,696 @@
+  let ready: Promise<void>
+
+  const launchWorker = (): Promise<void> => {
+    worker = new Worker(path)
+    return new Promise<void>((resolve, reject) => {
+      const msgHandler = (msg: any) => {
+        if (msg === 'ready') {
+          worker.off('error', reject)
+          worker.on('error', (e: any) => {
+            console.error(e, `Running worker ${basename(path)} terminated. Attempting relaunch...`)
+            worker.off('message', msgHandler)
+            // This won't result in an infinite loop because of exiting and
+            // because this handler is only executed after the 'ready' event
+            // Relaunch can happen multiple times, so long as the worker doesn't
+            // immediately fail.
+            ready = launchWorker().catch((e: any) => {
+              console.error(e, `Error on worker ${basename(path)} relaunch`)
+              process.exit(1)
+            })
+          })
+          resolve()
+        }
+      }
+      worker.on('message', msgHandler)
+      worker.once('error', reject)
+    })
+  }
+  ready = launchWorker()
+
+  const rpcSbp = (...args: any[]): Promise<any> => {
+    return ready.then(() => new Promise<any>((resolve, reject) => {
+      const mc = new MessageChannel()
+      const cleanup = ((worker: Worker) => () => {
+        worker.off('error', reject)
+        mc.port2.onmessage = null
+        mc.port2.onmessageerror = null
+      })(worker)
+      mc.port2.onmessage = (event) => {
+        cleanup()
+        const [success, result] = (event.data as any) as [boolean, any]
+        if (success) return resolve(result)
+        reject(result)
+      }
+      mc.port2.onmessageerror = () => {
+        cleanup()
+        reject(Error('Message error'))
+      }
+      worker.postMessage([mc.port1, ...args], [mc.port1 as any])
+      // If the worker itself breaks during an SBP call, we want to make sure
+      // this promise immediately rejects
+      worker.once('error', reject)
+    }))
+  }
+
+  return {
+    ready,
+    rpcSbp,
+    terminate: () => worker.terminate()
+  }
+}
+
+if (CREDITS_WORKER_TASK_TIME_INTERVAL && OWNER_SIZE_TOTAL_WORKER_TASK_TIME_INTERVAL > CREDITS_WORKER_TASK_TIME_INTERVAL) {
+  process.stderr.write('The size calculation worker must run more frequently than the credits worker for accurate billing')
+  process.exit(1)
+}
+
+// Temporarily disable workers in bundled environment
+// TODO: Implement proper worker bundling support
+const ownerSizeTotalWorker = undefined as WorkerType | undefined
+const creditsWorker = undefined as WorkerType | undefined
+
+// Node.js version 18 and lower don't have global.crypto defined
+// by default
+if (
+  !('crypto' in globalThis) &&
+  typeof require === 'function'
+) {
+  const crypto = await import('node:crypto')
+  const { webcrypto } = crypto
+  if (webcrypto) {
+    Object.defineProperty(globalThis, 'crypto', {
+      enumerable: true,
+      configurable: true,
+      get: () => webcrypto
+    })
+  }
+}
```

Comment:
I realise that this code _is_ in `server.js` currently, but IMHO it's something that we can now safely remove. We're no longer using Node 18 in GI, and we're not using Node 18 here.

Reply:
Removed.